controllers/hostingautoscalingpolicy/hostingautoscalingpolicy_controller.go

```diff
@@ -0,0 +1,450 @@
+
+		if scalableTargetDescription, err = ctx.ApplicationAutoscalingClient.DescribeScalableTargets(ctx, ResourceID); err != nil {
+			return scalableTargetDescriptionList, scalingPolicyDescriptionList, errors.Wrap(err, "Unable to describe ScalableTarget")
+		}
+
+		if scalingPolicyDescription, err = ctx.ApplicationAutoscalingClient.DescribeScalingPolicies(ctx, ctx.PolicyName, ResourceID); err != nil {
+			return scalableTargetDescriptionList, scalingPolicyDescriptionList, errors.Wrap(err, "Unable to describe ScalingPolicy")
+		}
+
+		if scalableTargetDescription != nil {
+			scalableTargetDescriptionList = append(ctx.ScalableTargetDescriptionList, scalableTargetDescription)
+		}
+		if scalingPolicyDescription != nil {
+			scalingPolicyDescriptionList = append(scalingPolicyDescriptionList, scalingPolicyDescription)
+		}
+	}
+	return scalableTargetDescriptionList, scalingPolicyDescriptionList, nil
+}
+
+// deleteAutoscalingPolicy converts Spec to Input, Registers Target, Creates second input, applies the scalingPolicy
+func (r *Reconciler) deleteAutoscalingPolicy(ctx reconcileRequestContext) error {
+	var deregisterScalableTargetInput applicationautoscaling.DeregisterScalableTargetInput
+	var deleteScalingPolicyInput applicationautoscaling.DeleteScalingPolicyInput
+
+	for _, ResourceID := range ctx.ResourceIDList {
+		deleteScalingPolicyInput = sdkutil.CreateDeleteScalingPolicyInput(ctx.HostingAutoscalingPolicy.Spec, ResourceID)
+		if _, err := ctx.ApplicationAutoscalingClient.DeleteScalingPolicy(ctx, &deleteScalingPolicyInput); err != nil && !clientwrapper.IsDeleteHAP404Error(err) {
+			return errors.Wrap(err, "Unable to DeleteScalingPolicy")
+		}
+	}
+
+	for _, ResourceID := range ctx.ResourceIDList {
+		deregisterScalableTargetInput = sdkutil.CreateDeregisterScalableTargetInput(ctx.HostingAutoscalingPolicy.Spec, ResourceID)
+		if _, err := ctx.ApplicationAutoscalingClient.DeregisterScalableTarget(ctx, &deregisterScalableTargetInput); err != nil && !clientwrapper.IsDeleteHAP404Error(err) {
+			return errors.Wrap(err, "Unable to DeregisterScalableTarget")
+		}
+	}
+
+	return nil
+}
+
+// applyAutoscalingPolicy converts Spec to Input, Registers Target, Creates scalingPolicy input, applies the scalingPolicy
+// This method needs to be broken down
+func (r *Reconciler) applyAutoscalingPolicy(ctx reconcileRequestContext) ([]*applicationautoscaling.DescribeScalableTargetsOutput, []*applicationautoscaling.ScalingPolicy, error) {
+	var registerScalableTargetInputList []applicationautoscaling.RegisterScalableTargetInput
+	var putScalingPolicyInputList []applicationautoscaling.PutScalingPolicyInput
+
+	// For describe
+	var scalableTargetDescriptionList []*applicationautoscaling.DescribeScalableTargetsOutput
+	var scalingPolicyDescriptionList []*applicationautoscaling.ScalingPolicy
+
+	// Review: Is this needed
+	if ctx.HostingAutoscalingPolicy.Spec.PolicyName == nil || len(*ctx.HostingAutoscalingPolicy.Spec.PolicyName) == 0 {
```

Comment:
We set this in `initializeContext` so this seems redundant?

Reply:
Agree, but Gautam recommended having it though he might have missed init. Not removing at the last minute now. Will change to TODO.